### DISCUSSION TITLE
java的sdk报错：empty grpc client: find no available rootcoord, check rootcoord state","status":1

### DISCUSSION THREAD

**Discussion by mmahao** [milvus-io/milvus#33812] [Q&A and General discussion]
java的sdk版本：2.2.13
milvus版本：2.2.16

突然sdk报错：

c.e.c.k.b.s.milvus.MilvusHandler -1 - {"exception":"java.lang.Exception: syncTimestamp Failed:stack trace: /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace\n/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:369 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).Call\n/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:382 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).ReCall\n/go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/client/client.go:424 github.com/milvus-io/milvus/internal/distributed/rootcoord/client.(*Client).AllocTimestamp\n/go/src/github.com/milvus-io/milvus/internal/proxy/timestamp.go:61 github.com/milvus-io/milvus/internal/proxy.(*timestampAllocator).alloc\n/go/src/github.com/milvus-io/milvus/internal/proxy/timestamp.go:83 github.com/milvus-io/milvus/internal/proxy.(*timestampAllocator).AllocOne\n/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:174 github.com/milvus-io/milvus/internal/proxy.(*baseTaskQueue).Enqueue\n/go/src/github.com/milvus-io/milvus/internal/proxy/impl.go:3063 github.com/milvus-io/milvus/internal/proxy.(*Proxy).Search\n/go/src/github.com/milvus-io/milvus/internal/distributed/proxy/service.go:876 github.com/milvus-io/milvus/internal/distributed/proxy.(*Server).Search\n/go/pkg/mod/github.com/milvus-io/milvus-proto/go-api/v2@v2.2.16/milvuspb/milvus.pb.go:11330 github.com/milvus-io/milvus-proto/go-api/v2/milvuspb._MilvusService_Search_Handler.func1: empty grpc client: find no available rootcoord, check rootcoord state","status":1}

没有找到具体的原因，猜测可能是因为混合查询之后，没有releaseCollection，导致内存不够用了，然后attu上的QueryCoord，IndexCoord，DataCoord全都没有了

如果先loadCollection，然后混合查询之后，不releaseCollection会怎么样？

**Comment by yhmo** [comment]
几个coordinator崩了，有可能是OOM，或者某些第三方组件(etcd, minio, pulsar)出问题导致。
loadCollection之后数据就保持在query node的内存里，直到releaseCollection之后才会把该表的数据清理出内存。不需要每次查询之后都调用releaseCollection清理。正常的做法就是load一次，然后查询无数次，等到不想再查询该表了，才调用releaseCollection请出内存。

**Reply by mmahao** [reply]
我这边是springboot，我是每次请求混合查询之前，都loadCollection了一次，但是混合查询之后，就没有再releaseCollection了，这种方式是不是不太合理？

是不是应该在sprintboot的服务启动完成的那一刻，只loadCollection一次，而不用每次请求进来的时候都loadCollection？

**Reply by yhmo** [reply]
loadCollection调用多次也没问题，它只会在第一次的时候把数据加载进内存，之后重复调用也只是立即返回，不会做什么事情。
服务端崩了一般是看崩之前的日志里有没有ERROR或者WARN。

**Reply by mmahao** [reply]
好的，谢谢
